libs/langchain/tests/unit_tests/schema/test_runnable.py

```diff
@@ -0,0 +1,515 @@
+        {"question": "What is your name?"},
+        {"question": "What is your favorite color?"},
+    ]
+    assert llm_spy.call_args.args[1] == [
+        ChatPromptValue(
+            messages=[
+                SystemMessage(content="You are a nice assistant."),
+                HumanMessage(content="What is your name?"),
+            ]
+        ),
+        ChatPromptValue(
+            messages=[
+                SystemMessage(content="You are a nice assistant."),
+                HumanMessage(content="What is your favorite color?"),
+            ]
+        ),
+    ]
+    assert tracer.runs == snapshot
+    mocker.stop(prompt_spy)
+    mocker.stop(llm_spy)
+
+    # Test stream
+    prompt_spy = mocker.spy(prompt.__class__, "ainvoke")
+    llm_spy = mocker.spy(llm.__class__, "astream")
+    tracer = FakeTracer()
+    assert [
+        token
+        async for token in chain.astream(
+            {"question": "What is your name?"}, dict(callbacks=[tracer])
+        )
+    ] == ["bar"]
+    assert prompt_spy.call_args.args[1] == {"question": "What is your name?"}
+    assert llm_spy.call_args.args[1] == ChatPromptValue(
+        messages=[
+            SystemMessage(content="You are a nice assistant."),
+            HumanMessage(content="What is your name?"),
+        ]
+    )
+
+
+@freeze_time("2023-01-01")
+def test_prompt_with_chat_model_and_parser(
+    mocker: MockerFixture, snapshot: SnapshotAssertion, fixed_uuids: None
+) -> None:
+    prompt = SystemMessage(content="You are a nice assistant.") + "{question}"
+    chat = FakeListChatModel(responses=["foo, bar"])
+    parser = CommaSeparatedListOutputParser()
+
+    chain = prompt | chat | parser
+
+    assert isinstance(chain, RunnableSequence)
+    assert chain.first == prompt
+    assert chain.middle == [chat]
+    assert chain.last == parser
+    assert dumps(chain, pretty=True) == snapshot
+
+    # Test invoke
+    prompt_spy = mocker.spy(prompt.__class__, "invoke")
+    chat_spy = mocker.spy(chat.__class__, "invoke")
+    parser_spy = mocker.spy(parser.__class__, "invoke")
+    tracer = FakeTracer()
+    assert chain.invoke(
+        {"question": "What is your name?"}, dict(callbacks=[tracer])
+    ) == ["foo", "bar"]
+    assert prompt_spy.call_args.args[1] == {"question": "What is your name?"}
+    assert chat_spy.call_args.args[1] == ChatPromptValue(
+        messages=[
+            SystemMessage(content="You are a nice assistant."),
+            HumanMessage(content="What is your name?"),
+        ]
+    )
+    assert parser_spy.call_args.args[1] == AIMessage(content="foo, bar")
+    assert tracer.runs == snapshot
+
+
+@freeze_time("2023-01-01")
+def test_seq_dict_prompt_llm(
+    mocker: MockerFixture, snapshot: SnapshotAssertion, fixed_uuids: None
+) -> None:
+    passthrough = mocker.Mock(side_effect=lambda x: x)
+
+    retriever = FakeRetriever()
+
+    prompt = (
+        SystemMessage(content="You are a nice assistant.")
+        + """Context:
```

Comment:
The operator overloaded semantics are non intuitive to me:


This works

```python
    prompt = (
        SystemMessage(content="You are a nice assistant.") + "blah"
    )
    chain = prompt | FakeListChatModel(responses=["foo, bar"])
```

This fails:

```python
    prompt = (
        SystemMessage(content="You are a nice assistant.")
    )
    chain = prompt | FakeListChatModel(responses=["foo, bar"])
```

The `+` on the messages changes the underlying type to something that's a runnable object.